eclair-core/src/main/scala/fr/acinq/eclair/payment/offer/OfferManager.scala

```diff
@@ -0,0 +1,331 @@
+            val signedMetadata = SignedMetadata(nodeKey, invoiceRequest.offer.offerId, metadata)
+            val pathId = signedMetadataCodec.encode(signedMetadata).require.bytes
+            val receivePayment = MultiPartHandler.ReceiveOfferPayment(nodeKey, invoiceRequest, routes, router, preimage, pathId, additionalTlvs, customTlvs)
+            val child = context.spawnAnonymous(CreateInvoiceActor(nodeParams))
+            child ! CreateInvoiceActor.CreateBolt12Invoice(context.messageAdapter[Try[Bolt12Invoice]](WrappedInvoice), receivePayment)
+            waitForInvoice(replyPath, postman)
+        }
+      })
+    }
+
+    private def waitForInvoice(replyPath: OnionMessages.Destination,
+                               postman: ActorRef[Postman.SendMessage]): Behavior[Command] = {
+      Behaviors.setup(context => {
+        Behaviors.receiveMessagePartial {
+          case WrappedInvoice(Success(invoice)) =>
+            postman ! Postman.SendMessage(Nil, replyPath, None, TlvStream(OnionMessagePayloadTlv.Invoice(invoice.records)), context.messageAdapter[Postman.OnionMessageResponse](WrappedOnionMessageResponse), 0 seconds)
+            waitForSent()
+          case WrappedInvoice(_) =>
+            Behaviors.stopped
+        }
+      })
+    }
+
+    private def waitForSent(): Behavior[Command] = {
+      Behaviors.receiveMessagePartial {
+        case WrappedOnionMessageResponse(_) =>
+          Behaviors.stopped
+      }
+    }
+  }
+
+  object PaymentActor {
+    sealed trait Command
+
+    /**
+     * Sent by the offer handler. Causes the creation of a dummy invoice that matches as best as possible the actual invoice for this payment (since the actual invoice is not stored) and will be used in the payment handler.
+     * @param additionalTlvs additional TLVs to add to the dummy invoice. Should be the same as what was used for the actual invoice.
+     * @param customTlvs     custom TLVs to add to the dummy invoice. Should be the same as what was used for the actual invoice.
+     */
+    case class AcceptPayment(additionalTlvs: Seq[InvoiceTlv] = Nil, customTlvs: Seq[GenericTlv] = Nil) extends Command
+
+    /**
+     * Sent by the offer handler to reject the payment. For instance because stock has been exhausted.
+     */
+    object RejectPayment extends Command
+
+    def apply(nodeParams: NodeParams, replyTo: ActorRef[MultiPartHandler.GetIncomingPaymentActor.Command], offer: Offer, metadata: PaymentMetadata): Behavior[Command] = {
+      Behaviors.receiveMessage {
+        case AcceptPayment(additionalTlvs, customTlvs) =>
+          val dummyKey = PublicKey(hex"020000000000000000000000000000000000000000000000000000000000000001")
+          // This invoice is not the one we've sent (we don't store the real one as it would be a DoS vector) but it shares all the important bits with the real one.
+          val dummyInvoice = Bolt12Invoice(TlvStream(offer.records.records ++ Seq[InvoiceTlv](
```

Comment:
This isn't very satisfying, we're forced to include ~200 bytes of dummy fields (because they are required fields for a valid Bolt12 invoice) and the resulting invoice will look weird when displayed.

I'm not 100% sure it's a good idea, but have you looked into defining a `case class PaidBolt12Invoice(records: TlvStream[InvoiceTlv]) extends Invoice` that wouldn't have the requirements of `Bolt12Invoice` and would let us include only the fields that we really need to store? We can define a separate `val hrp = "lnpi"` for it to make the serialization more explicit (which would be future-proof if at some point we start storing "real" Bolt12 invoices in the DB)?